app/src/Controllers/ZipController.php

```diff
@@ -34,41 +34,161 @@ public function __invoke(Request $request, Response $response): ResponseInterfac
             return $response->withStatus(404, $this->translator->trans('error.file_not_found'));
         }
 
-        $response->getBody()->write(
-            $this->cache->get(sprintf('zip-%s', sha1($path)), function () use ($path): string {
-                return $this->createZip($path)->getContents();
-            })
-        );
-
-        return $response->withHeader('Content-Type', 'application/zip')
+        $response = $response
+            ->withHeader('Content-Type', 'application/zip')
             ->withHeader('Content-Disposition', sprintf(
                 'attachment; filename="%s.zip"',
                 $this->generateFileName($path)
-            ));
+            ))
+            ->withHeader('X-Accel-Buffering', 'no');
```

Comment:
What is the purpose of the `X-Accel-Buffering` header here?

Reply:
This is to prevent Nginx (and I think any FastCGI server) from buffering the response: https://maennchen.dev/ZipStream-PHP/guide/Nginx.html